python/cudf/reshape/general.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+import numpy as np
+
+from cudf.dataframe import Series
+from cudf.dataframe import Buffer
+from cudf.dataframe import DataFrame
+from cudf.utils import cudautils
+from cudf.dataframe.categorical import CategoricalColumn
+
+
+def melt(frame, id_vars=None, value_vars=None, var_name='variable',
+         value_name='value'):
+    """Unpivots a DataFrame from wide format to long format,
+    optionally leaving identifier variables set.
+    
+    Parameters
+    ----------
+    frame : DataFrame
+    id_vars : tuple, list, or ndarray, optional
+        Column(s) to use as identifier variables.
+        default: None
+    value_vars : tuple, list, or ndarray, optional
+        Column(s) to unpivot.
+        default: all columns that are not set as `id_vars`.
+    var_name : scalar
+        Name to use for the `variable` column.
+        default: frame.columns.name or 'variable'
+    value_name : str
+        Name to use for the `value` column.
+        default: 'value'
+
+    Returns
+    -------
+    molten : DataFrame
+
+    Difference from pandas:
+     * Does not support 'col_level' because cuDF does not have multi-index
+
+    TODO: Examples
+    """
+
+    # Arg cleaning
+    import types
+    # id_vars
+    if id_vars is not None:
+        if not isinstance(id_vars, list):
+            id_vars = [id_vars]
+        missing = set(id_vars) - set(frame.columns)
+        if not len(missing) == 0:
+            raise KeyError(
+                "The following 'id_vars' are not present"
+                " in the DataFrame: {missing}"
+                "".format(missing=list(missing)))
+    else:
+        id_vars = []
+
+    # value_vars
+    if value_vars is not None:
+        if not isinstance(value_vars, list):
```

Comment:
I think this should be `collections.abc.sequence` to allow for tuple and list-like objects.

Reply:
done and tested